pygen/pygen_src/riscv_instr_gen_config.py

```diff
@@ -12,23 +12,260 @@
+        self.mstatus_mprv = 0
+        self.mstatus_mxr = 0
+        self.mstatus_sum = 0
+        self.mstatus_tvm = 0
+        self.mstatus_fs = BitArray(bin(0b0), length=2)
+        self.mstatus_vs = BitArray(bin(0b0), length=2)
+        self.mtvec_mode = list(map(lambda vec_md: vec_md.name, mtvec_mode_t))
+        self.fcsr_rm = list(map(lambda csr_rm: csr_rm.name, f_rounding_mode_t))
         self.enable_sfence = 0
-        self.reserved_regs = []
+        # dicts for exception_cause_t & interrupt_cause_t Enum classes
+        self.m_mode_exception_delegation = {}
+        self.s_mode_exception_delegation = {}
+        self.m_mode_interrupt_delegation = {}
+        self.s_mode_interrupt_delegation = {}
+        self.support_supervisor_mode = 0
+        self.no_wfi = 1
+        self.min_stack_len_per_program = 10 * (stng.XLEN / 8)
+        self.max_stack_len_per_program = 16 * (stng.XLEN / 8)
+        self.stack_len = 5000
+        self.use_push_data_section = 0
+        self.signature_addr = 0xdeadbeef
+        self.require_signature_addr = 0
+        self.main_program_instr_cnt = 0  # count of main_prog
+        self.instr_cnt = 200
+        self.num_of_tests = 1
+        self.no_data_page = 0
+        self.no_branch_jump = 0
+        self.no_load_store = 0
+        self.no_csr_instr = 0
+        self.no_ebreak = 1
+        self.no_dret = 1
+        self.no_fence = 1
+        self.enable_unaligned_load_store = 0
+        self.illegal_instr_ratio = 0
+        self.hint_instr_ratio = 0
+        self.num_of_harts = stng.NUM_HARTS
+        self.tvec_alignment = 2
+        self.fix_sp = 0
+        self.mode_opts = "boot_mode_opts"
+        self.enable_page_table_exception = 0
+        self.enable_nested_interrupt = 0
+        self.test_suffix = "asm_test_suffix"
+        self.enable_interrupt = 0
+        self.no_directed_instr = 0
+        self.enable_timer_irq = 0
+        self.bare_program_mode = 0
         self.enable_illegal_csr_instruction = 0
         self.enable_access_invalid_csr_level = 0
+        self.enable_misaligned_instr = 0
+        self.enable_dummy_csr_write = 0
+        self.randomize_csr = 0
+        self.allow_sfence_exception = 0
+        self.no_delegation = 1
+        self.force_m_delegation = 0
+        self.force_s_delegation = 0
+        self.support_supervisor_mode = 0
+        self.disable_compressed_instr = 0
+        self.gen_debug_section = 0
+        self.enable_ebreak_in_debug_rom = 0
+        self.set_dcsr_ebreak = 0
+        self.num_debug_sub_program = 0
+        self.enable_debug_single_step = 0
+        self.single_step_iterations = 0
+        self.set_mstatus_tw = 0
+        self.set_mstatus_mprv = 0
+        self.max_branch_step = 20
+        self.max_directed_instr_stream_seq = 20
+        self.enable_floating_point = 0
+        self.enable_vector_extension = 0
+        self.enable_b_extension = 0
+        self.enable_bitmanip_groups = ['ZBB', 'ZBS', 'ZBP', 'ZBE', 'ZBF',
```

Comment:
Why you do not create enum for this one?

Reply:
Thanks for the suggestion, I expected the list might change values for different config, hence retained the SV logic. If it remains constant it can moved to enum class.